idea/testData/inspectionsLocal/collections/uselessCallOnCollection/MapNotNullWithLambda5.kt

```diff
@@ -0,0 +1,14 @@
+// WITH_RUNTIME
+// FIX: Change call to 'map'
+
+fun test(): List<String> {
+    return listOf(1, 2, 3).<caret>mapNotNull { i ->
+        foo {
+            bar(i)
+        }
+    }
+}
+
+fun <T> foo(f: () -> T): T = f()
+
+fun bar(i: Int): String = ""
```

Comment:
@igoriakovlev 
`MapNotNullWithLambda5.kt` test fails In latest master(069adebf0102d5b9f6fe5929e902eda816c608f2).

Reply:
@t-kameyama 
I mean that I do not see this FP in last master IDE. Did you checked it?

Reply:
I checked IDE behaviour, found false negative in latest master(5efbe6ae).

In lastest master:
```kotlin
fun test() {
    // Not suggested. It's OK
    val a: List<String> = listOf(1, 2, 3).mapNotNull { i ->
        foo {
            bar(i)
        }
    }

    // Not suggested. It's false negative.
    val b: List<String> = listOf(1, 2, 3).mapNotNull { i ->
        foo {
            baz(i)
        }
    }
}

fun <T> foo(f: () -> T): T = f()

fun bar(i: Int): String? = null

fun baz(i: Int): String = ""
```


In this branch:
```kotlin
fun test() {
    // Not suggested It's OK.
    val a: List<String> = listOf(1, 2, 3).mapNotNull { i ->
        foo {
            bar(i)
        }
    }

    // Suggested. It's OK.
    val b: List<String> = listOf(1, 2, 3).mapNotNull { i ->
        foo {
            baz(i)
        }
    }
}

fun <T> foo(f: () -> T): T = f()

fun bar(i: Int): String? = null

fun baz(i: Int): String = ""
```

Should I close this pull request and create separate issue?